app/src/main/java/com/annhienktuit/cleanarchitectureplayer/ui/player/PlayerActivity.java

```diff
@@ -7,32 +7,56 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
-import com.annhienktuit.cleanarchitectureplayer.DefaultServiceLocator;
+import com.annhienktuit.cleanarchitectureplayer.MainThreadExecutorService;
 import com.annhienktuit.cleanarchitectureplayer.R;
-import com.annhienktuit.cleanarchitectureplayer.ServiceLocator;
+import com.annhienktuit.cleanarchitectureplayer.di.components.DaggerApplicationComponent;
+import com.annhienktuit.cleanarchitectureplayer.di.components.ApplicationComponent;
+import com.annhienktuit.cleanarchitectureplayer.di.modules.AppModule;
+import com.annhienktuit.domain.interfaces.SongDataSource;
 import com.annhienktuit.domain.usecases.GetSongUseCase;
 import com.google.android.exoplayer2.ExoPlayer;
 import com.google.android.exoplayer2.source.DefaultMediaSourceFactory;
 import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
 
+import java.util.concurrent.ExecutorService;
+
+import javax.inject.Inject;
+
 public class PlayerActivity extends AppCompatActivity implements PlayerView {
 
     public static final String SONG_ID = "id";
 
     private com.google.android.exoplayer2.ui.PlayerView playerView;
 
-    private PlayerPresenter presenter;
+    private PlayerPresenterInterface presenter;
 
```

Comment:
interface should be xxxPresenter, implementation of it should be xxxPresenterImpl